spec/components/Web3API_Client.md

```diff
@@ -0,0 +1,157 @@
+# Web3API Client
+
+Web3API client is intended for application developers that can use the client as a single library in their (d)apps to interact with various decentralized services using Web3API's packages.
+
+Currently, Web3API client is available in the following languages:
+ - JavaScript (TypeScript)
+
+## Client interface
+The following snippet contains the interface that client should implement.
+
+```
+interface Web3APIClient {
+  redirects(): UriRedirect[];
+  query(options: QueryApiOptions): QueryApiResult;
+}
+
+type UriRedirect {
+ // Redirect from this URI or set of URIs specified by the regex
+ from: Uri | RegExp;
```

Comment:
I guess string?

Reply:
Oh I see Uri interface down below. You should put it here before first usage or link it.

Reply:
I don't think markdown has so many options for linking xD Moving.